india_compliance/gst_india/doctype/gst_settings/gst_settings.py

```diff
@@ -346,6 +348,65 @@ def has_valid_credentials(self, gstin, service, throw=False):
 
         return True
 
+    @frappe.whitelist()
+    def check_gst_account_changes(self):
+        previous_doc = self.get_doc_before_save()
+        old_gst_accounts = {item.name: item for item in previous_doc.gst_accounts}
+        new_gst_accounts = {item.name: item for item in self.gst_accounts}
+
+        if len(new_gst_accounts) != len(old_gst_accounts):
+            return True
+
+        for account_name in new_gst_accounts:
+            old_account = old_gst_accounts.get(account_name)
+
+            if self.has_row_changed(old_account, new_gst_accounts[account_name]):
+                return True
+
+        return False
+
+    def has_row_changed(self, old_gst_account, new_gst_account):
+        if old_gst_account is None:
+            return True
+
+        return any(
+            old_gst_account.get(field) != new_gst_account.get(field)
+            for field in ["company", "account_type", *GST_ACCOUNT_FIELDS]
+        )
+
+
+@frappe.whitelist()
+def schedule_gst_patches(cron_time):
+    dt = datetime.strptime(cron_time, "%Y-%m-%d %H:%M:%S")
```

Comment:
No need for execution time. Whitelist for permissions. GST Settings edit or Systems Manager.